test/jdk/java/io/Reader/Of.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.Reader;
+import java.io.StringReader;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.ReadOnlyBufferException;
+
+import org.testng.annotations.*;
+
+import static org.testng.Assert.*;
+
+/*
+ * @test
+ * @bug 8341566
+ * @run testng Of
+ * @summary Check for expected behavior of Reader.of().
+ */
+public class Of {
```

Comment:
Would it be possible to include a test which creates and operates on a `Reader` which is backed by a `CharSequence` of zero length. Similarly, a test which creates and operates on a `Reader` which is backed by a `CharSequence` of `Integer.MAX_VALUE` length? I think that should give us good coverage for the implementation of the returned `Reader` for these corner cases.

Reply:
Zero-case is partically covered by the existing test code. Your request looks like a real lot of work, as it implies a complete rewrite of the overall structure of the test - and frankly spoken, my budget for this issue is near to spent already. Does that *really* provide a justifying benefit, given the fact that these are literally edge cases (and given the fact, that the code is a 1:1 copy of `StringReader`, which we all used since decades, including those edge cases)? If you think it is really necessary, can you please post the modified version of `Of.java` here? I will then post it into the PR. Thanks.